plugins/outputs/http/http.go

```diff
@@ -0,0 +1,210 @@
+	"github.com/influxdata/telegraf/plugins/serializers/json"
+	"net"
+	"net/http"
+	"strings"
+	"time"
+)
+
+var sampleConfig = `
+  ## It requires a url name.
+  ## Will be transmitted telegraf metrics to the HTTP Server using the below URL.
+  ## Note that not support the HTTPS.
+  url = "http://127.0.0.1:8080/metric"
+  ## http_headers option can add a custom header to the request.
+  ## The value is written as a delimiter(:).
+  ## Content-Type is required http header in http plugin.
+  ## so content-type of HTTP specification (plain/text, application/json, etc...) must be filled out.
+  http_headers = [ "Content-Type:application/json" ]
+  ## With this HTTP status code, the http plugin checks that the HTTP request is completed normally.
+  ## As a result, any status code that is not a specified status code is considered to be an error condition and processed.
+  expected_status_codes = [ 200, 204 ]
+  ## Configure response header timeout in seconds. Default : 3
+  response_header_timeout = 3
+  ## Configure dial timeout in seconds. Default : 3
+  dial_timeout = 3
+
+  ## Data format to output.
+  ## Each data format has it's own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_OUTPUT.md
+  data_format = "influx"
+`
+
+const (
+	POST = "POST"
+
+	DEFAULT_RESPONSE_HEADER_TIMEOUT = 3
+	DEFAULT_DIAL_TIME_OUT           = 3
+)
+
+type Http struct {
+	// http required option
+	URL            string   `toml:"url"`
+	HttpHeaders    []string `toml:"http_headers"`
+	ExpStatusCodes []int    `toml:"expected_status_codes"`
+
+	// Option with http default value
+	ResHeaderTimeout int `toml:"response_header_timeout"`
+	DialTimeOut      int `toml:"dial_timeout"`
+
+	client     http.Client
+	serializer serializers.Serializer
+	// expStatusCode that stores option values received with expected_status_codes
+	expStatusCode map[int]bool
+}
+
+func (h *Http) SetSerializer(serializer serializers.Serializer) {
+	h.serializer = serializer
+}
+
+// Connect to the Output
+func (h *Http) Connect() error {
+	h.client = http.Client{
+		Transport: &http.Transport{
+			Dial: (&net.Dialer{
+				Timeout: time.Duration(h.DialTimeOut) * time.Second,
+			}).Dial,
+			ResponseHeaderTimeout: time.Duration(h.ResHeaderTimeout) * time.Second,
+		},
+	}
+
+	return nil
+}
+
+// Close is not implemented. Because http.Client not provided connection close policy. Instead, uses the response.Body.Close() pattern.
+func (h *Http) Close() error {
+	return nil
+}
+
+// Description A plugin that can transmit metrics over HTTP
+func (h *Http) Description() string {
+	return "A plugin that can transmit metrics over HTTP"
+}
+
+// SampleConfig provides sample example for developer
+func (h *Http) SampleConfig() string {
+	return sampleConfig
+}
+
+// Writes metrics over HTTP POST
+func (h *Http) Write(metrics []telegraf.Metric) error {
+	if err := validate(h); err != nil {
+		return err
```

Comment:
@Shugyousha Oh, It's interesting. You are right. 
Every time the write function is called, the validation check is done continuously since the beginning. 
If the output interface has Validate, check the required option in the individual Outputs in advance, and then it will be very good if the setting is wrong. 
Until then, I will remove the corresponding check.
Thanks a lot. 👍